src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Batch.cs

```diff
@@ -473,6 +498,20 @@ private void ServerMessageHandler(object sender, SqlInfoMessageEventArgs args)
             }
         }
 
+        /// <summary>
+        /// Aggregates all result sets in the batch into a single special action 
+        /// </summary>
+        private SpecialAction processResultSetSpecialActions()
+        {
+            SpecialAction batchSpecialAction = new SpecialAction();
```

Comment:
Create an instance from this in the class to use, no need to create it in the method